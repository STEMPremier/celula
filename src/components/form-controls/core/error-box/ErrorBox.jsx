import React from 'react';
import PropTypes from 'prop-types';

import './error-box.less';

const ErrorBox = ({ errorMsg = '' }) => (
  <div className="ce-errorbox">
    <div className="ce-errorbox__container">
      <div className="ce-errorbox__arrow" />
      <div className="ce-errorbox__message">
        <span className="ce-errorbox__text">{errorMsg}</span>
      </div>
    </div>
  </div>
);

export default ErrorBox;

ErrorBox.propTypes = {
  /**
   * An error message to display
   */
  errorMsg: PropTypes.string.isRequired,
};
