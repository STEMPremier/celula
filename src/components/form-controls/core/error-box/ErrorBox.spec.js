/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import ErrorBox from '.';

describe('<ErrorBox />', () => {
  describe('render()', () => {
    test('render the ErrorBox component', () => {
      const wrapper = shallow(
        <ErrorBox errorMsg="This is an error message." />,
      );

      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});

/* eslint-enable react/jsx-filename-extension */
