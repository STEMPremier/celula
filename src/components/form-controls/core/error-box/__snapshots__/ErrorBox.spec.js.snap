// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorBox /> render() render the ErrorBox component 1`] = `
<div
  className="ce-errorbox"
>
  <div
    className="ce-errorbox__container"
  >
    <div
      className="ce-errorbox__arrow"
    />
    <div
      className="ce-errorbox__message"
    >
      <span
        className="ce-errorbox__text"
      >
        This is an error message.
      </span>
    </div>
  </div>
</div>
`;
